### PR TITLE
Fix Doxygen warnings in goto-programs/ folder

### PR DESCRIPTION
**Pull request by johnnonweiler**
Update function parameter documentation, or rename parameters, to fix Doxygen warnings

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/goto-programs/goto_convert.cpp [hunk 1]
```diff
@@ -1974,6 +1974,7 @@ void goto_convert(
 /// \param thread_body: sequence of codet's that can be executed
 ///   in parallel.
 /// \param [out] dest: resulting goto-instructions.
+/// \param mode: language mode.
```
Comment (peterschrammel) [inline src/goto-programs/goto_convert.cpp]: `Language mode` (no full stop unless it is a full sentence)

**Review by peterschrammel — APPROVED** [review]

**Review by NlightNFotis — APPROVED** [review]
LGTM

**Review by chrisr-diffblue — APPROVED** [review]

File: src/goto-programs/goto_convert.cpp (hunk 1 above)
Comment (johnnonweiler) [reply src/goto-programs/goto_convert.cpp]: I was being consistent with the other parameter documentation for this method.  I find changing style within the documentation for one function is much more off-putting than having a different style between functions.

I thought people would prefer me not to change the documentation for the other parameters to match the coding standard.  Would you prefer me to do that?

**Comment by johnnonweiler** [comment]
Following discussion with @peterschrammel, I have now changed the parameter documentation for `goto_convert` to match the coding standard (by starting with capital letters and _not_ finishing with a full stop).

**Review by chrisr-diffblue — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: f1e368e967b05e33df7bde83bd5f5ac5fac6769b).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/91239488
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Comment by johnnonweiler** [comment]
I believe the test-gen bot failure was a spurious failure because "the cbmc commit has disappeared in the mean time".

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: dcd6cd15e8fc277136948fa26c3a0a8f9b28ef15).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/91244913

**Pull request merged**